hello-world/blazor/README.md

```diff
@@ -0,0 +1,356 @@
+
+    try {
+        if (!cvRouter?.disposed) {
+            await cvRouter?.dispose();
+        }
+        if (!cameraEnhancer?.disposed) {
+            await cameraEnhancer?.dispose();
+        }
+
+        // Reset components
+        cameraViewContainer.style.display = "none";
+        cameraViewContainer.innerHTML = "";
+        resultsContainer.textContent = "";
+    } catch (ex) {
+        let errMsg = ex.message || ex;
+        console.error(errMsg);
+    }
+}
+```
+
+### Create a `Pages\DecodeImage.razor` component
+
+In `Pages\DecodeImage.razor`, we will modify the component to enable barcode decoding through an image. This page will be accessible at the `/image` URL.
+
+```razor
+@page "/image"
+@inject IJSRuntime JS
+
+<PageTitle>Dynamsoft Barcode Reader Hello World - Blazor</PageTitle>
+
+<h1>Decode Image</h1>
+<InputFile id="inputElement" style="margin-bottom: 1rem" OnChange="DecodeImageTask" accept="image/*" />
+<h3 style="margin-bottom: 1rem">Results:</h3>
+<div id="results" style="width: 100%; height: 30vh; overflow: auto; white-space: pre-wrap"></div>
+
+@code {
+    // reference: https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-javascript-from-dotnet?view=aspnetcore-8.0#invoke-js-functions
+
+    private async Task DecodeImageTask(InputFileChangeEventArgs e)
+    {
+        // Dynamically load decode image module
+        await JS.InvokeVoidAsync("loadDecodeImageModule");
+
+        var imageFile = e.File;
+        var jsImageStream = imageFile.OpenReadStream(1024 * 1024 * 20);
+        var dotnetImageStream = new DotNetStreamReference(jsImageStream);
+
+        await JS.InvokeAsync<string>("startImageDecode");
+    }
+
+    public void Dispose()
+    {
+        JS.InvokeAsync<string>("cleanUpImageDecode");
+    }
+}
+```
+
+> Note:
+>
+> We can significantly improve the initial load speed and performance when we dynamically import Dynamsoft's Barcode Scanning component on-demand. In the sample code above, we dynamically imported the decode image module through the line `await JS.InvokeVoidAsync("loadDecodeImageModule");`.
+
+### Create a `wwwroot\decodeImage.js` file
+
+In `wwwroot\decodeImage.js`, we will instantiate `cvRouter` to decode barcodes in images. 
+
+These functions will be called by the Blazor components through JavaScript Interop.
+
+Reference: https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-javascript-from-dotnet?view=aspnetcore-8.0#invoke-js-functions
+
+```
+// Create JS function "startImageDecode"
+window.startImageDecode = async () => {
+    const inputElement = document.getElementById("inputElement");
+    const resultsContainer = document.getElementById("results");
+
+    const file = inputElement.files[0]; // Get the first file from the input element
+
+    try {
+        if (file) {
+            // Convert file to blob
+            const arrayBuffer = await file.arrayBuffer();
+            const blob = new Blob([arrayBuffer]);
```

Comment:
`file.instanceOf(Blob) === true`, so do not need to convert.

Reply:
After get the file, you can reset `inputElement.value`, so user can test a image repeatedly.

Reply:
Agree. thank you